.github/workflows/playwright.yml

```diff
@@ -1,24 +1,47 @@
 name: Playwright Tests
 on:
   push:
-    branches: [ main ]
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
 jobs:
   test:
-    timeout-minutes: 60
+    timeout-minutes: 10
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - uses: actions/setup-node@v4
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
       with:
-        node-version: lts/*
+        node-version: 20.12.2
     - name: Install dependencies
-      run: npm ci
+      run: |
+        npm ci
+        npm install wait-on
     - name: Install Playwright Browsers
       run: npx playwright install --with-deps
-    - name: Run Playwright tests
-      run: npx playwright test
-    - uses: actions/upload-artifact@v4
+    - name: Build the app
+      run: npm run build
+    - name: Start Vite server and run tests
+      env: 
+        PLAYWRIGHT_BASE_URL: http://localhost:5173
+      run: |
+        # Start Vite dev server in background
```

Comment:
This is the secret I was missing before: background the dev server with `&` and then run the tests.